File: NeuroamWPF/Neuroam/Neuroam/source/IO/JsonFile.cs
using System;
using System.IO;
using Newtonsoft.Json;

namespace Neuroam.IO
{
    public class JsonFile
    {
        string m_FileName;
        public JsonFile(string fileName)
        {
            m_FileName = fileName;

            try
            {
                FileStream stream = File.Open(m_FileName, FileMode.OpenOrCreate);
                stream.Close();
            }
            catch (Exception e)
            {
                Logger.Instance.LogException($"Failed to open/create file {m_FileName}.", e);
            }
        }

        public string ReadAll()
        {
            return File.ReadAllText(m_FileName);
        }

        public void WriteAll(string data)
        {
            File.WriteAllText(m_FileName, data);
        }
    }
}

File: Neuroam/source/IO/JsonFile.cpp
#include <PrecompiledHeader.h>

#include <IO/JsonFile.h>

namespace Neuroam {

JsonFile::JsonFile(StringView fileName)
{
    m_FileName = fileName;
}

StringView JsonFile::ReadAll()
{
    return "";
}

void JsonFile::WriteAll(StringView)
{
}

}